### PR TITLE
IDEM-303

### PR DESCRIPTION
**Pull request by JasminFarhi**
IMG_7897

### CODE REVIEW

**Review by bitcoinbrisbane — APPROVED** [review]

File: app/src/navigation/screens/ClaimScreen.tsx [hunk 1]
```diff
@@ -112,17 +112,19 @@ const ClaimScreen: React.FC = () => {
...
       if (!isEmail(formState.email as string)) {
+        setLoading(false);
         return Alert.alert(
           AlertTitle.Warning,
           "Please type a valid email claim value in the input field."
         );
       }
-      setLoading(false);
+    }
+    await addClaim(claim.type, formState, selectedFileIds);
```
Comment (brettvniekerk) [inline app/src/navigation/screens/ClaimScreen.tsx:124]: Could u not put this in the block above, considering they both check for `if (isEmailClaim)` ?

**Review by brettvniekerk — CHANGES_REQUESTED** [review]
One minor thing but otherwise LGTM

File: app/src/navigation/screens/ClaimScreen.tsx (hunk 1 above)
Comment (JasminFarhi) [reply app/src/navigation/screens/ClaimScreen.tsx:124]: Not at the moment with how the logic is working.

**Review by amagon270 — APPROVED** [review]

File: app/src/navigation/screens/ClaimScreen.tsx [hunk 2]
```diff
@@ -112,17 +112,19 @@ const ClaimScreen: React.FC = () => {
...
+    if (isEmailClaim) {
       const email = (formState.email as string).toLowerCase();
-      await verifyPublicKey(email);
```
[the remark this answers is not in the document]
Comment (amagon270) [reply app/src/navigation/screens/ClaimScreen.tsx:125]: as it stands it shouldn't affect the flow

**Pull request merged**